support-lib/jni/djinni_support.cpp

```diff
@@ -239,8 +239,13 @@ jint JniEnum::ordinal(JNIEnv * env, jobject obj) const {
 
 LocalRef<jobject> JniEnum::create(JNIEnv * env, jint value) const {
     LocalRef<jobject> values(env, env->CallStaticObjectMethod(m_clazz.get(), m_staticmethValues));
+    jniExceptionCheck(env);
     DJINNI_ASSERT(values, env);
-    return LocalRef<jobject>(env, env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()), value));
+    LocalRef<jobject> result(env,
+                             env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()),
+                                                        value));
+    jniExceptionCheck(env);
+    return std::move(result);
```

Comment:
Ok, I did some more research and I stand corrected. Making it const doens't prevent RVO, however in cases where the compiler cannot apply RVO (it has other prerequisites) it is forced to *copy* the return value instead of *move* it.

But RVO is still just an optimization so the code must be well-formed without it. That means making a local of move-only type const prevents you from returning it because it cannot be moved from, making the code ill-formed in the first place. RVO is not allowed to make that magically compile.